django/db/models/query.py

```diff
@@ -941,12 +949,16 @@ def get_or_create(self, defaults=None, **kwargs):
                     pass
                 raise
 
+    get_or_create.alters_data = True
```

Comment:
`alters_data` was not not set on `get_or_create()` or `get_or_update()` since their introduction (b3a6348bc24de6dc7168e6c00f2d2d440b1fe748). I can imagine that somewhere in the universe :vulcan_salute: it's used on templates to fetch objects since they don't always create a new one. IMO, we should at least add a backward incompatible note.

Reply:
Good point - pushed up a release note :+1: